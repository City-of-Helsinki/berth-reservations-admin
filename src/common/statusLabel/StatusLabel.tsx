import React from 'react';
import classNames from 'classnames';
import { StatusLabel as HDSStatusLabel, StatusLabelProps as HDSStatusLabelProps } from 'hds-react';

import styles from './statusLabel.module.scss';

export interface StatusLabelProps {
  type: HDSStatusLabelProps['type'] | 'warning';
  label: string;
  className?: string;
}

const StatusLabel = ({ type, label, className }: StatusLabelProps) => {
  const statusLabelType = type === 'warning' ? 'alert' : type;

  return (
    <HDSStatusLabel className={classNames({ [styles.warning]: type === 'warning' }, className)} type={statusLabelType}>
      {label}
    </HDSStatusLabel>
  );
};

export default StatusLabel;
