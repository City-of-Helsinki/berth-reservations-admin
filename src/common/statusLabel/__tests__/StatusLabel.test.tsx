import React from 'react';
import { shallow } from 'enzyme';
import StatusLabel, { StatusLabelProps } from '../StatusLabel';

describe('StatusLabel', () => {
  const getWrapper = (props: StatusLabelProps = { label: 'Foo', type: 'alert' }) => shallow(<StatusLabel {...props} />);

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });
});
