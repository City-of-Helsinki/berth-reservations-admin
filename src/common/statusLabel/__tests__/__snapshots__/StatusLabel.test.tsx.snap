// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusLabel renders normally 1`] = `
<span
  class="StatusLabel-module_statusLabel__3R2J2 status-label_hds-status-label__1L8YI StatusLabel-module_alert__2VR8r status-label_hds-status-label--alert__1ecsX"
>
  Foo
</span>
`;
